test/jsonrpc-test.js

```diff
@@ -74,14 +74,21 @@ module.exports = {
       testBadRequest = function (testJSON, done) {
         var req = new MockRequest('POST');
         var res = new MockResponse();
-        server.handleHttp(req, res);
-        req.emit('data', testJSON);
-        req.emit('end');
-        var decoded = JSON.parse(res.httpBody);
-        expect(decoded.id).to.equal(null);
-        expect(decoded.error.message).to.equal('Invalid Request');
-        expect(decoded.error.code).to.equal(-32600);
-        done();
+        server.handleHttp(req, res).then(function (result) {
+          console.log(result);
+        }).catch(function (err) {
+          expect(err.id).to.equal(undefined);
+          expect(err.message).to.equal('Invalid Request');
+          expect(err.code).to.equal(-32600);
+          done();  
```

Comment:
every done when using promises could do inside `catch(...).finally(done)`, if it throws, mocha will catch it (not really important, but to make it easier for debugging)